hub/util/get_storage_provider.py

```diff
@@ -66,6 +66,7 @@ def storage_provider_from_path(
     elif path.startswith("hub://"):
         storage = storage_provider_from_hub_path(path, read_only, token=token)
     else:
+        path = os.path.expanduser(path)
```

Comment:
**Bonus bugfix**: Support `~` prefixed paths

Reply:
Is the main difference between del ds and ds.delete() that ds.delete() doesn't kill the name, so you can still technically access the variable "ds"

Reply:
@istranic yes. `del` is a special python keyword that removes the name from the scope, so future references to it will throw a NameError as if it wasn't defined in the first place.